static/src/js/util/accessMethods/buildAccessMethods.js

```diff
@@ -25,202 +19,61 @@ export const buildAccessMethods = (collectionMetadata, isOpenSearch) => {
-          description,
-          enableTemporalSubsetting: true,
-          enableSpatialSubsetting: true,
-          hierarchyMappings,
-          id: serviceConceptId,
-          isValid: true,
-          keywordMappings,
-          longName,
-          name,
-          supportedOutputFormats: uniq(outputFormats),
-          supportedOutputProjections: outputProjections,
-          supportsBoundingBoxSubsetting: supportsBoundingBoxSubsetting(serviceItem),
-          supportsShapefileSubsetting: supportsShapefileSubsetting(serviceItem),
-          supportsTemporalSubsetting: supportsTemporalSubsetting(serviceItem),
-          supportsVariableSubsetting: supportsVariableSubsetting(serviceItem),
-          supportsConcatenation: supportsConcatenation(serviceItem),
-          defaultConcatenation: defaultConcatenation(serviceItem),
-          enableConcatenateDownload: defaultConcatenation(serviceItem),
-          type: serviceType,
-          url: urlValue,
-          variables
-        }
+  const nonDownloadMethods = serviceItems !== null ? serviceItems.reduce((methods, serviceItem) => {
+    let associatedVariables = collectionAssociatedVariables
+    const {
+      type: serviceType,
+      variables: serviceAssociatedVariables = {}
+    } = serviceItem
 
-        harmonyIndex += 1
-      }
+    // Overwrite variables if there are variables associated to the service record
+    if (serviceAssociatedVariables.items && serviceAssociatedVariables.items.length > 0) {
+      associatedVariables = serviceAssociatedVariables
+    }
 
-      if (serviceType.toLowerCase() === 'swodlr' && (disableSwodlr !== 'true')) {
-        accessMethods.swodlr = {
-          id: serviceConceptId,
-          isValid: true,
-          longName,
-          name,
-          type: serviceType,
-          supportsSwodlr: true,
-          url: urlValue
-        }
-      }
-    })
-  }
+    const lowerServiceType = serviceType.toLowerCase()
 
-  // Determine if the collection should have the downloadable accessMethod
-  let onlineAccessFlag = false
+    // Only process service types that EDSC supports
+    const supportedServiceTypes = ['esi', 'echo orders', 'opendap', 'harmony', 'swodlr']
+    if (!supportedServiceTypes.includes(lowerServiceType)) return {}
 
-  if (granules) {
-    // If the collection has granules, check their online access flags to
-    // determine if this collection is downloadable
-    const { items: granuleItems } = granules
+    const params = {
+      disableOrdering,
+      harmonyIndex,
+      associatedVariables,
+      disableSwodlr
+    }
 
-    if (granuleItems) {
-      onlineAccessFlag = isDownloadable(granuleItems)
+    const updatedMethods = {
+      ...methods,
+      ...buildMethods[lowerServiceType](serviceItem, params)
     }
 
-    if (onlineAccessFlag || isOpenSearch) {
-      accessMethods.download = {
-        isValid: true,
-        type: 'download'
-      }
+    if (lowerServiceType === 'harmony') {
+      harmonyIndex += 1
```

Comment:
ESI and ECHO Orders should be refactored to use this method of indexing their access methods